src/FrameBuffer.cpp

```diff
@@ -708,6 +717,14 @@ void FrameBufferList::attachDepthBuffer()
 	if (m_pCurrent == nullptr)
 		return;
 
+	//Needed by some devices
+	const GLenum discards[]  = {GL_DEPTH_ATTACHMENT};
+#ifdef GLES2
+	glDiscardFramebufferEXT(GL_FRAMEBUFFER, 1, discards);
+#else
+	glInvalidateFramebuffer(GL_FRAMEBUFFER, 1, discards);
+#endif
+
```

Comment:
It is wrong functionality. If it fixes issues with some devices, enable it for these devices only.